packages/api/lib/rulesHelpers.js

```diff
@@ -531,21 +531,21 @@ async function buildPayload(item) {
   return {
     template,
     definition,
-    provider: item.provider,
-    collection: item.collection,
-    meta: get(item, 'meta', {}),
-    cumulus_meta: get(item, 'cumulus_meta', {}),
-    payload: get(item, 'payload', {}),
-    queueUrl: item.queueUrl,
-    asyncOperationId: item.asyncOperationId,
-    executionNamePrefix: item.executionNamePrefix,
+    provider: rule.provider,
+    collection: rule.collection,
```

Comment:
These properties don't exist on a Postgres rule, as far as I know? There would only be `rule.provider_cumulus_id` and `rule.collection_cumulus_id`

Reply:
That's a really good point. I've updated the buildPayload function to translate the PG rule, since the translate functions makes a call to look up the provider and collection [`bb74e3a` (#2871)](https://github.com/nasa/cumulus/pull/2871/commits/bb74e3aa1d4642235116c170a15beaa6bb63eb93#diff-803bdbf08060a60307d82cb3938c90378a3c359858b4decb1d9089444c92d57bR523)

Reply:
Sorry, there were a couple changes I made after that last commit https://github.com/nasa/cumulus/pull/2871/files#diff-803bdbf08060a60307d82cb3938c90378a3c359858b4decb1d9089444c92d57bR519 The gist is that I updated buildPayload to take in an ApiRule and all calls to `buildPayload` need to translate a PG rule to an API rule. Also, I made `asyncOperationId` and `cumulus_meta` fields optional parameters to the function.

Reply:
I proposed a refactor to the logic here:

https://github.com/nasa/cumulus/pull/2871/files?show-viewed-files=true&file-filters%5B%5D=#r821139656